src/ext/java/org/opentripplanner/ext/siri/AffectsMapper.java

```diff
@@ -91,117 +94,105 @@ private List<EntitySelector> mapVehicleJourneys(AffectsScopeStructure.VehicleJou
         }
       }
 
+      LocalDate serviceDate = null;
+
+      if (affectedVehicleJourney.getOriginAimedDepartureTime() != null) {
+        ZonedDateTime startOfService = ServiceDateUtils.asStartOfService(
+          affectedVehicleJourney.getOriginAimedDepartureTime()
+        );
+
+        serviceDate = startOfService.toLocalDate();
+      }
+
       List<VehicleJourneyRef> vehicleJourneyReves = affectedVehicleJourney.getVehicleJourneyReves();
 
       if (isNotEmpty(vehicleJourneyReves)) {
+        List<FeedScopedId> affectedTripIds = new ArrayList<>();
+
         for (VehicleJourneyRef vehicleJourneyRef : vehicleJourneyReves) {
           List<FeedScopedId> tripIds = new ArrayList<>();
 
-          FeedScopedId tripIdFromVehicleJourney = getTripId(
+          TripOnServiceDate tripOnServiceDate = entityResolver.resolveTripOnServiceDate(
```

Comment:
Is there a reason to not just use `entityResolver.resolveTrip` here? Now trips which do not have a DSJ are not resolved.

Reply:
Updated to use `entityResolver.resolveTrip`

The service date is used to apply the PtSituation only on trips with the specified service date - e.g. if the trip passes midnight.

Reply:
Yes, but the service date is necessarily not the on the date of the departure from the first stop.

Reply:
Unfortunately, the legacy-hack included defining the service date in the field `OriginAimedDepartureTime` - in lack of alternative fields, so it is not _actually_ interpreted as the departuretime from the first stop.